[libmariadb]:x64-mingw-static build error

**Host Environment**
- OS: Windows
Hi, i'm trying to generate cmake cache using vcpkg and x64-mingw-static but this error occurs every time, I don't know what to do
I can't generate cache with x64-windows because it needs an visual studio instance, and I don't want to use vs
I know x64-mingw-static is a community triplet configuration but maybe someone can fix this
**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install libmariadb[core,mariadbclient,schannel,ssl]:x64-mingw-static`

**Failure logs**
-- Building x64-mingw-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:129 (message):
    Command failed: C:/opentibia/CMake/bin/cmake.exe --build . --config Debug --target install -- -v -j7
    Working Directory: C:/opentibia/vcpkg/buildtrees/libmariadb/x64-mingw-static-dbg
    See logs for more information:
      C:\opentibia\vcpkg\buildtrees\libmariadb\install-x64-mingw-static-dbg-out.log

Call Stack (most recent call first):
  C:/opentibia/canary/build/windows-release/vcpkg_installed/x64-mingw-static/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  C:/opentibia/canary/build/windows-release/vcpkg_installed/x64-mingw-static/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/libmariadb/portfile.cmake:62 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building libmariadb:x64-mingw-static failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libmariadb
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[libmariadb]+Build+error
Include '[libmariadb] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-07-21-a0e87e227afb536c62188c11ad029954f28fdb22
    vcpkg-scripts version: 47a0bd967 2022-07-28 (10 hours ago)

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

@Thomas1664 thank you it solved the issue, but I have one on luajit now, can you help me with that?
as far as i understand the installation is getting the wrong Makefile, it should be using src/makefile(for mingw compilation), i don't know exactly

Installing 1/1 luajit:x64-mingw-static...
Building luajit[core]:x64-mingw-static...
warning: -- Using community triplet x64-mingw-static. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\opentibia\vcpkg\triplets\community\x64-mingw-static.cmake
-- Using cached LuaJIT-LuaJIT-v2.0.5.tar.gz.
-- Cleaning sources at C:/opentibia/vcpkg/buildtrees/luajit/src/v2.0.5-86b97027bc.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/opentibia/vcpkg/downloads/LuaJIT-LuaJIT-v2.0.5.tar.gz
-- Applying patch 001-fix-build-path.patch
-- Applying patch 002-fix-crt-linkage.patch
-- Applying patch 003-do-not-set-macosx-deployment-target.patch
-- Using source at C:/opentibia/vcpkg/buildtrees/luajit/src/v2.0.5-86b97027bc.clean
-- Building x64-mingw-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:129 (message):
    Command failed: make -j7 -f C:/opentibia/vcpkg/buildtrees/luajit/src/v2.0.5-86b97027bc.clean/Makefile clean
    Working Directory: C:/opentibia/vcpkg/buildtrees/luajit/src/v2.0.5-86b97027bc.clean
    See logs for more information:
      C:\opentibia\vcpkg\buildtrees\luajit\clean-x64-mingw-static-debug-out.log
      C:\opentibia\vcpkg\buildtrees\luajit\clean-x64-mingw-static-debug-err.log

Call Stack (most recent call first):
  ports/luajit/portfile.cmake:41 (vcpkg_execute_build_process)
  scripts/ports.cmake:147 (include)


error: building luajit:x64-mingw-static failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+luajit
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[luajit]+Build+error
Include '[luajit] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-07-21-a0e87e227afb536c62188c11ad029954f28fdb22
    vcpkg-scripts version: 66045de4d 2022-07-29 (22 hours ago)

You can also use the prefilled template from C:\opentibia\vcpkg\installed\vcpkg\issue_body.md.

--- a/ports/libmariadb/portfile.cmake
+++ b/ports/libmariadb/portfile.cmake
@@ -1,13 +1,9 @@
-if (EXISTS "${CURRENT_INSTALLED_DIR}/share/libmysql")
+if(EXISTS "${CURRENT_INSTALLED_DIR}/share/libmysql")
     message(FATAL_ERROR "FATAL ERROR: libmysql and libmariadb are incompatible.")
 endif()
 
 if("openssl" IN_LIST FEATURES AND "schannel" IN_LIST FEATURES)
     message(FATAL_ERROR "Only one SSL backend must be selected.")
-endif()
-
-if("schannel" IN_LIST FEATURES AND NOT VCPKG_TARGET_IS_WINDOWS)
-    message(FATAL_ERROR "Feature schannel not supported on non-Windows platforms.")
 endif()
 
 vcpkg_from_github(
@@ -29,7 +25,7 @@ vcpkg_from_github(
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
-    FEATURES 
+    FEATURES
         iconv WITH_ICONV
         mariadbclient VCPKG_MARIADBCLIENT
 )
@@ -46,7 +42,7 @@ vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         ${FEATURE_OPTIONS}
-        -DINSTALL_INCLUDEDIR=include/mysql  # legacy port decisiong
+        -DINSTALL_INCLUDEDIR=include/mysql # legacy port decisiong
         -DINSTALL_LIBDIR=lib
         -DINSTALL_PLUGINDIR=plugins/${PORT}
         -DWITH_UNIT_TESTS=OFF
@@ -69,9 +65,9 @@ if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
         vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/libmariadb.pc" " -lmariadb" " -llibmariadb")
     endif()
 endif()
+
 vcpkg_fixup_pkgconfig()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-# copy license file
 file(INSTALL "${SOURCE_PATH}/COPYING.LIB" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/libmariadb/vcpkg.json
+++ b/ports/libmariadb/vcpkg.json
@@ -47,17 +47,17 @@
           "name": "libmariadb",
           "default-features": false,
           "features": [
-            "schannel"
+            "openssl"
           ],
-          "platform": "windows & !mingw"
+          "platform": "!windows | mingw"
         },
         {
           "name": "libmariadb",
           "default-features": false,
           "features": [
-            "openssl"
+            "schannel"
           ],
-          "platform": "!windows | mingw"
+          "platform": "windows & !mingw"
         }
       ]
     },

--- a/ports/libmariadb/vcpkg.json
+++ b/ports/libmariadb/vcpkg.json
@@ -1,8 +1,10 @@
 {
   "name": "libmariadb",
   "version-semver": "3.1.15",
+  "port-version": 1,
   "description": "MariaDB Connector/C is used to connect C/C++ applications to MariaDB and MySQL databases",
   "homepage": "https://github.com/MariaDB/mariadb-connector-c",
+  "license": "LGPL-2.1-only",
   "supports": "!uwp",
   "dependencies": [
     {
@@ -35,7 +37,8 @@
       ]
     },
     "schannel": {
-      "description": "SSL support (Secure Channel)"
+      "description": "SSL support (Secure Channel)",
+      "supports": "windows & !mingw"
     },
     "ssl": {
       "description": "Default SSL backend",
@@ -46,7 +49,7 @@
           "features": [
             "schannel"
           ],
-          "platform": "windows | mingw"
+          "platform": "windows & !mingw"
         },
         {
           "name": "libmariadb",
@@ -54,7 +57,7 @@
           "features": [
             "openssl"
           ],
-          "platform": "!windows & !mingw"
+          "platform": "!windows | mingw"
         }
       ]
     },

--- a/ports/libmariadb/vcpkg.json
+++ b/ports/libmariadb/vcpkg.json
@@ -4,7 +4,7 @@
   "port-version": 1,
   "description": "MariaDB Connector/C is used to connect C/C++ applications to MariaDB and MySQL databases",
   "homepage": "https://github.com/MariaDB/mariadb-connector-c",
-  "license": "LGPL-2.1-only",
+  "license": "LGPL-2.1-or-later",
   "supports": "!uwp",
   "dependencies": [
     {
@@ -38,7 +38,7 @@
     },
     "schannel": {
       "description": "SSL support (Secure Channel)",
-      "supports": "windows & !mingw"
+      "supports": "windows | mingw"
     },
     "ssl": {
       "description": "Default SSL backend",
@@ -49,7 +49,7 @@
           "features": [
             "openssl"
           ],
-          "platform": "!windows | mingw"
+          "platform": "!windows & !mingw"
         },
         {
           "name": "libmariadb",
@@ -57,7 +57,7 @@
           "features": [
             "schannel"
           ],
-          "platform": "windows & !mingw"
+          "platform": "windows | mingw"
         }
       ]
     },

--- a/ports/libmariadb/vcpkg.json
+++ b/ports/libmariadb/vcpkg.json
@@ -47,17 +47,17 @@
           "name": "libmariadb",
           "default-features": false,
           "features": [
-            "openssl"
+            "schannel"
           ],
-          "platform": "!windows & !mingw"
+          "platform": "windows | mingw"
         },
         {
           "name": "libmariadb",
           "default-features": false,
           "features": [
-            "schannel"
+            "openssl"
           ],
-          "platform": "windows | mingw"
+          "platform": "!windows & !mingw"
         }
       ]
     },

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3814,7 +3814,7 @@
     },
     "libmariadb": {
       "baseline": "3.1.15",
-      "port-version": 0
+      "port-version": 1
     },
     "libmaxminddb": {
       "baseline": "1.4.3",

--- a/versions/l-/libmariadb.json
+++ b/versions/l-/libmariadb.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "12e3ba664a0a66a9b549d112630bd5488193bb42",
+      "version-semver": "3.1.15",
+      "port-version": 1
+    },
+    {
       "git-tree": "c51d4e78081ee708c1cdebeaaa201aa0064079e3",
       "version-semver": "3.1.15",
       "port-version": 0

--- a/versions/l-/libmariadb.json
+++ b/versions/l-/libmariadb.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "12e3ba664a0a66a9b549d112630bd5488193bb42",
+      "git-tree": "06122c8b00cc582d3ee36072fcbd03fa9385f238",
       "version-semver": "3.1.15",
       "port-version": 1
     },

